openapi.yaml

```diff
@@ -269,6 +269,91 @@ paths:
               schema:
                 $ref: '#/components/schemas/Problem'
 
+  /share:
+    post:
+      summary: Save the given AsyncAPI document.
+      operationId: saveDoc
+      tags:
+        - share
+      requestBody:
+        description: Parse your AsyncAPI document.
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ShareRequest"
+      responses:
+        "200":
+          description: AsyncAPI document successfully saved.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ShareResponse"
+        "400":
+          description: The given AsyncAPI document(s) is/are not valid.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Problem"
+        "422":
+          description: The given AsyncAPI document(s) is/are not valid due to invalid parameters in the request.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Problem"
+        default:
+          description: Unexpected problem.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Problem'
+  /share/{id}:
+    get:
+      summary: Retrieve a given AsyncAPI document
+      operationId: getDoc
+      tags:
+        - share
+      parameters:
+        - name: id
+          in: path
+          required: true
+          schema: 
+            $ref: "#/components/schemas/RetrieveRequest" 
```

Comment:
If you define `id` as parameter, then validation schema for it should point to the parameter, not to the parameters "object", so `RetrieveRequest` schema should look like:

```yaml
type: string
description: Document unique identifier
```

Also we should make such a schema inline - please remove `RetrieveRequest` schema from `components.schemas` section.